### ISSUE TITLE
Inverting a CompoundModel of RotateNative2Celestial looses units

### ISSUE THREAD

**Issue by Cadair** [astropy/astropy#8800]
I noticed this bug just before I did a live demo of some gWCS stuff, it only got my pixel coordinates wrong by about 6000 pixels :laughing: 

It seems that when creating a CompoundModel with `RotateNative2Celestial` the Parameter converts the arcsecond value into radians, and then back to degrees but still thinks the unit is arcseconds.

```python
from astropy.modeling.models import RotateNative2Celestial, Pix2Sky_TAN
import astropy.units as u


model = RotateNative2Celestial(lon=414.12767466*u.arcsec, lat=482.43170875*u.arcsec, lon_pole=180.*u.deg)
print("Original lon, lat")
print(model)
```
```
Model: RotateNative2Celestial
Inputs: ('phi_N', 'theta_N')
Outputs: ('alpha_C', 'delta_C')
Model set size: 1
Parameters:
        lon          lat      lon_pole
       arcsec       arcsec      deg
    ------------ ------------ --------
    414.12767466 482.43170875    180.0
```
```python
inverse = model.inverse
print("Inverse lon, lat")
print(inverse)
```
```
Model: RotateCelestial2Native
Inputs: ('alpha_C', 'delta_C')
Outputs: ('phi_N', 'theta_N')
Model set size: 1
Parameters:
        lon          lat      lon_pole
       arcsec       arcsec      deg
    ------------ ------------ --------
    414.12767466 482.43170875    180.0

```
```python
model = model | Pix2Sky_TAN()
print("Compound lon, lat")
print(model)
```
```
Model: CompoundModel0
Inputs: ('phi_N', 'theta_N')
Outputs: ('phi', 'theta')
Model set size: 1
Expression: [0] | [1]
Components:
    [0]: <RotateNative2Celestial(lon=414.12767466 arcsec, lat=482.43170875 arcsec, lon_pole=180. deg)>

    [1]: <Pix2Sky_Gnomonic()>
Parameters:
           lon_0               lat_0        lon_pole_0
           arcsec              arcsec          deg
    ------------------- ------------------- ----------
    0.11503546518333334 0.13400880798611112      180.0
```
```python

inverse = model.inverse
print("Inverse lon, lat")
print(inverse)
```
```
Model: CompoundModel1
Inputs: ('phi', 'theta')
Outputs: ('phi_N', 'theta_N')
Model set size: 1
Expression: [0] | [1]
Components:
    [0]: <Sky2Pix_Gnomonic()>

    [1]: <RotateCelestial2Native(lon=0.11503547 arcsec, lat=0.13400881 arcsec, lon_pole=180. deg)>
Parameters:
            lon_1                 lat_1         lon_pole_1
            arcsec                arcsec           deg
    --------------------- --------------------- ----------
    3.195429588425926e-05 3.722466888503087e-05      180.0
```

**Comment by nden** [comment]
@Cadair I spent a couple of hours chasing this bug and it is due again to the disconnect between the parameters of the compound model and the parameters of its constituent models. Essentially the problem is that not all the information of the original parameters gets copied to the parameters of the compound model.  There's no easy way to fix it. Even if it's possible it is going to be another hack. 

Instead of fixing it I suggest we work on getting #8769 reviewed and merged because this kind of problems are solved there by design.

**Comment by Cadair** [comment]
>  I spent a couple of hours chasing this bug

Well it probably would have taken me a week :laughing: Thanks for chasing it down.

Seems reasonable to fix it with #8769 I will give that branch a try out when I am back in the UK and see if I can break that one :stuck_out_tongue:

**Comment by nden** [comment]
@Cadair I think this was fixed by #8769.
```
model = RotateNative2Celestial(lon=414.12767466*u.arcsec, lat=482.43170875*u.arcsec, lon_pole=180.*u.deg) | Pix2Sky_TAN()

model.inverse(*model(1,2))                                                                               
Out[13]: (0.9999999999999997, 2.0)
```

I am going to close it but please reopen if you think there's still something wrong.
